examples/iga/galerkin_laplace_problem.py

```diff
@@ -85,7 +86,7 @@ def map_positions(positions, x_min, x_max, y_min, y_max):
 # Assemble individual elements
 ukv = solution_field.unique_knots
 n_dofs = solution_field.control_points.shape[0]
-system_matrix = np.zeros((n_dofs, n_dofs))
+system_matrix = lil_matrix((n_dofs, n_dofs))
```

Comment:
> Intended Usage
>LIL is a convenient format for constructing sparse matrices
>
>once a matrix has been constructed, convert to CSR or CSC format for fast arithmetic and matrix vector operations
>
>consider using the COO format when constructing large matrices

There is no conversion

Reply:
By putting the system matrix together in a for loop, I think the lil_matrix would be better suited for that.
The problem with the coo_matrix is that it doesn't support slicing directly:

> Disadvantages of the COO format
> does not directly support:
- > arithmetic operations
- > slicing
> 
If we want to use coo_matrix to improve performance, the code gets a bit more complex.

Reply:
I don't say we need to use coo_sparse matrices right away. I am suggesting, that we convert the matrix once the assembly is done in order to use it more efficiently in matrix arithmetics as required for solving